src/SmartComponents/BaselinesPage/EditBaseline/DeleteFactModal/DeleteFactModal.js

```diff
@@ -0,0 +1,174 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { Modal, Button } from '@patternfly/react-core';
+
+import { editBaselineActions } from '../redux';
+import editBaselineHelpers from '../helpers';
+
+class DeleteFactModal extends Component {
+    constructor(props) {
+        super(props);
+        this.deleteFacts = this.deleteFacts.bind(this);
+
+        this.state = {
+            modalOpened: this.props.modalOpened
+        };
+    }
+
+    async deleteFacts() {
+        const { patchBaseline, fetchBaselineData, baselineData, editBaselineTableData } = this.props;
+        let apiBody = editBaselineHelpers.makeDeleteFactsPatch(editBaselineTableData, baselineData);
+
+        let results = await patchBaseline(baselineData.id, apiBody);
+        if (results) {
+            fetchBaselineData(baselineData.id);
+        }
+
+        this.props.toggleModal();
+    }
+
+    buildMessage = (categoryMessage, factMessage) => {
+        let numFactsMessage;
+        let deleteCatMessage;
+        let message;
+
+        if (categoryMessage && factMessage) {
+            numFactsMessage = 'You have selected ' + categoryMessage + ' and ' + factMessage + ' to be deleted.';
+            deleteCatMessage = 'Deleting a category will delete all facts within the category.';
+            message = <div>
+                <span className='display-block'>{ numFactsMessage }</span>
+                <span className='display-block'>{ deleteCatMessage }</span>
+            </div>;
+        } else if (categoryMessage) {
+            numFactsMessage = 'You have selected ' + categoryMessage + ' to be deleted.';
+            deleteCatMessage = 'Deleting a category will delete all facts within the category.';
+            message = <div>
+                <span className='display-block'>{ numFactsMessage }</span>
+                <span className='display-block'>{ deleteCatMessage }</span>
+            </div>;
+        } else if (factMessage) {
+            numFactsMessage = 'You have selected ' + factMessage + ' to be deleted.';
+            message = <div>
+                <span className='display-block'>{ numFactsMessage }</span>
+            </div>;
+        }
+
+        return message;
+    }
+
+    countFacts = () => {
```

Comment:
For me, this is an unituitive name for what it returns. This function both counts facts and returns a message about them. The counting could be moved out to a helper for easier testing and this function focus on generating the message inside the component.